### PR TITLE
Change max sql threads based on number of consumers

### PR DESCRIPTION
**Pull request by mohitkhullar**
R7 merge for https://github.com/bloomberg/comdb2/pull/1722/files

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
@@ sqlite/src/parse.y @@
+ Custom tokens and rules must be defined after SQLite's.
+ Please update SQL documentation and bubble diagrams to reflect the changes you've made, if necessary.
@@ sqlite/src/vdbeInt.h @@
+ If the order of members in struct Mem is changed, columnNullValue (defined in sqlite/vdbeapi.c) needs to be reordered accordingly.
@@ sqlite/src/vdbeaux.c @@
+ If a new fixed-length serial type is introduced, sqlite3IsFixedLengthSerialType (defined in sqlite/vdbeInt.h) needs to be changed accordingly.
@@ db/prefault_helper.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/sqlglue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/fdb_bend.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/handle_buf.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ schemachange/sc_global.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/walkback.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/logmsg.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
```

File: util/thdpool.c [hunk 1]
```diff
@@ -1023,6 +1027,14 @@ int thdpool_get_nbusythds(struct thdpool *pool)
     return pool->thdlist.count - pool->freelist.count;
 }
 
+void thdpool_add_waitthd(struct thdpool *pool) {
+     pool->nwaitthd++;
```
Comment (nirbhayc) [inline util/thdpool.c:1031]: Use atomic add here; and we wouldn't need consumer_sqlthds_mutex.

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/2cdf953d/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/340 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>datetime</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/2cdf953d/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/2cdf953d/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2740'>Resubmit</a></p>

**Review by nirbhayc — APPROVED** [review]
Use ATOMIC_ADD(1/-1) instead. Looks good otherwise.

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
@@ sqlite/src/parse.y @@
+ Custom tokens and rules must be defined after SQLite's.
+ Please update SQL documentation and bubble diagrams to reflect the changes you've made, if necessary.
@@ sqlite/src/vdbeInt.h @@
+ If the order of members in struct Mem is changed, columnNullValue (defined in sqlite/vdbeapi.c) needs to be reordered accordingly.
@@ sqlite/src/vdbeaux.c @@
+ If a new fixed-length serial type is introduced, sqlite3IsFixedLengthSerialType (defined in sqlite/vdbeInt.h) needs to be changed accordingly.
@@ db/prefault_helper.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/sqlglue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/fdb_bend.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/handle_buf.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ schemachange/sc_global.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/walkback.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/logmsg.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/3f4c44f2/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/3f4c44f2/smoketest.log'>Error &#x26A0; </a>.<br>Failed to compile.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/3f4c44f2/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/3f4c44f2/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2740'>Resubmit</a></p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
@@ sqlite/src/parse.y @@
+ Custom tokens and rules must be defined after SQLite's.
+ Please update SQL documentation and bubble diagrams to reflect the changes you've made, if necessary.
@@ sqlite/src/vdbeInt.h @@
+ If the order of members in struct Mem is changed, columnNullValue (defined in sqlite/vdbeapi.c) needs to be reordered accordingly.
@@ sqlite/src/vdbeaux.c @@
+ If a new fixed-length serial type is introduced, sqlite3IsFixedLengthSerialType (defined in sqlite/vdbeInt.h) needs to be changed accordingly.
@@ db/prefault_helper.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/sqlglue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/fdb_bend.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/handle_buf.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ schemachange/sc_global.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/walkback.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/logmsg.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/89d19351/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/340 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>systbl_proc_leak<br>datetime</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/89d19351/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/89d19351/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2740'>Resubmit</a></p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
@@ sqlite/src/parse.y @@
+ Custom tokens and rules must be defined after SQLite's.
+ Please update SQL documentation and bubble diagrams to reflect the changes you've made, if necessary.
@@ sqlite/src/vdbeInt.h @@
+ If the order of members in struct Mem is changed, columnNullValue (defined in sqlite/vdbeapi.c) needs to be reordered accordingly.
@@ sqlite/src/vdbeaux.c @@
+ If a new fixed-length serial type is introduced, sqlite3IsFixedLengthSerialType (defined in sqlite/vdbeInt.h) needs to be changed accordingly.
@@ db/prefault_helper.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/sqlglue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/fdb_bend.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/handle_buf.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ schemachange/sc_global.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/walkback.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/logmsg.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/89d19351/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/340 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>datetime</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/89d19351/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/89d19351/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2740'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/89d19351/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/340 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>datetime</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/89d19351/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/89d19351/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2740'>Resubmit</a></p>

**Pull request merged**